src/Faithlife.WebRequests/Json/GenericStatusCodeResponse.cs

```diff
@@ -0,0 +1,32 @@
+using System.Net;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Faithlife.WebRequests.Json
+{
+	/// <summary>
+	/// AutoWebServiceRequest responses with an explicitly set status code. 
+	/// </summary>
+	public class GenericStatusCodeResponse : AutoWebServiceResponse
+	{
+		/// <summary>
+		/// Exposes the response status code.
+		/// </summary>
+		public HttpStatusCode StatusCode { get; internal set; } // internal instead of private; reflection can't get private setters of derived classes
+
+		/// <summary>
+		/// Exposes the response headers.
+		/// </summary>
+		public HttpResponseHeaders Headers { get; private set; }
+
+		/// <summary>
+		/// Sets Headers property.
+		/// </summary>
+		/// <param name="info">The web service response handler information.</param>
+		protected override async Task OnResponseHandledCoreAsync(WebServiceResponseHandlerInfo info)
+		{
+			await base.OnResponseHandledCoreAsync(info).ConfigureAwait(false);
+			Headers = info.WebResponse.Headers;
```

Comment:
Instead of overriding this method, should we use a `private protected m_responseHeaders` in the base class (and expose it here)?

Reply:
Looks like doing so would require upgrading the project to use C# 7.2. Do we want that?

Reply:
I don't have a problem with using C# 7.2, but I don't think we should make Bradley's proposed change. Fields should be private.

Reply:
Yes; it was a mistake to suggest a field; it should be a `private protected` property that this class could expose.

There is definitely some overhead in `await`ing a base method, which could be avoided by not having this override; OTOH, this is making a web service call so a few extra microseconds in an extra `await` probably aren't going to be noticeable.

Reply:
^ this is not intended to be read as rejecting the current approach; simply correcting my previous comment